matchbox_socket/src/ggrs_socket.rs

```diff
@@ -7,17 +7,10 @@ impl WebRtcSocket {
     #[must_use]
     pub fn players(&self) -> Vec<PlayerType<String>> {
         // needs to be consistent order across all peers
-        let mut ids = self.connected_peers();
-        ids.push(self.id().to_owned());
-        ids.sort();
-        ids.iter()
-            .map(|id| {
-                if id == self.id() {
-                    PlayerType::Local
-                } else {
-                    PlayerType::Remote(id.to_owned())
-                }
-            })
+        self.connected_peers()
+            .iter()
+            .map(|id| PlayerType::Remote(id.clone()))
+            .chain([PlayerType::Local].into_iter())
```

Comment:
Ah, this is the hard challenge (and the reason I left it as a todo instead of just doing it).

In ggrs, the order needs to be the same across all peers (hence the sort, before we replaced it with `PlayerType::Local`.

So I would think this breaks `matchbox_demo`?

To solve it, I think we would need to either:

A. bring back `WebRtcSocket::id` as an `Option<PeerId>`,
B. reconsider making socket creation async... though makes the socket slightly more cumbersome to use from bevy... but might be worth it.

I think perhaps A is goods as a first step, then we can gather and discuss the pros and cons of a async connect-to-signalling-server method in public API in another issue.

Reply:
Ah, good catch, I hadn't noticed the same player was being moved by both clients :laughing: 

I've went ahead and implemented option A, however I'm not too happy synchronously access the mutex, any ideas?
Reckon async socket creation is the way to go in the long run, with the possibility of providing a way of sending user setup data